ProviderFactory/GeoIP2Factory.php

```diff
@@ -29,7 +29,9 @@ protected function getProvider(array $config): Provider
     {
         $provider = $config['provider'];
         if ('webservice' === $provider) {
-            $provider = new Client($config['user_id'], $config['license_key'], $config['locales'], $config['webservice_options']);
+            $userId = is_string($config['user_id']) ? (int) $config['user_id'] : $config['user_id'];
```

Comment:
This looks strange. But I understand that you dont want to cast null to int. 
How about: 

```
$userId = isset($config['user_id']) ? (int) $config['user_id'] : null;
```

Reply:
Updated. :)